Objects/unionobject.c

```diff
@@ -422,6 +422,30 @@ static PyNumberMethods union_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static const char* const cls_attrs[] = {
+        "__name__",
+        "__qualname__",
```

Comment:
We only need ``__module__`` right? I'm -1 on adding ``__name__`` and ``__qualname__``. A union object is not a callable (so it's definitely not a class, function, method, descriptor, or generator instance), so this doesn't make a lot of sense to me.

https://docs.python.org/3/library/stdtypes.html#definition.__name__

Similarly, ``typing.Union`` does not have ``__name__``, only the ABCs. Eg. ``Union[int, str].__Name__ is None``.

WDYT?

Reply:
Yup, totally agree. We don't need `__qualname__` and `__name__` to fix `typing.Annotated` compatability.